examples/nova_cell_custom/kustomization.yaml

```diff
@@ -0,0 +1,33 @@
+apiVersion: kustomize.config.k8s.io/v1alpha1
+kind: Component
+
+namespace: openstack
+
+components:
+- ../preprovisioned # for baremetal nodes, replace with baremetal
+
+resources:
+- openstackdataplanenovaservice.yaml
+
+configMapGenerator:
+- name: nova-extra-config
+  files:
+    - 25-nova-extra.conf=nova-extra.conf
+  options:
+    disableNameSuffixHash: true
+
+patches:
+- target:
+    kind: OpenStackDataPlaneService
+    name: nova
```

Comment:
kustomize will just use the service as a base, patch it and then create nova-cell-custom

Reply:
Right, so we're kustomizing the name in kustomization.yaml so it won't get overwritten during service reconcile.

I agree though, this is confusing as `openstackdataplanenovaservice.yaml` is just a copy of `config/services/dataplane_v1beta1_openstackdataplaneservice_nova.yaml`. Now we'd have to maintain that in 2 places with this change.

This would be more powerful if we could figure out how to run this example directly against `config/services/dataplane_v1beta1_openstackdataplaneservice_nova.yaml`, and then document for a user how they would do that, b/c that is the most likely workflow for a user.

When I tried this diff:

```
Mon Mar 25 04:26:58 (jslagle@teletran-1) [~/code/osp-ng/openstack-k8s-operators/dataplane-operator/examples/nova_cell_custom] (kustomizeupdate)$ git diff
diff --git a/examples/nova_cell_custom/kustomization.yaml b/examples/nova_cell_custom/kustomization.yaml
index cb20d6b..441c4f0 100644
--- a/examples/nova_cell_custom/kustomization.yaml
+++ b/examples/nova_cell_custom/kustomization.yaml
@@ -7,7 +7,7 @@ components:
 - ../preprovisioned # for baremetal nodes, replace with baremetal
 
 resources:
-- openstackdataplanenovaservice.yaml
+- ../../config/services/dataplane_v1beta1_openstackdataplaneservice_nova.yaml
 
 configMapGenerator:
 - name: nova-extra-config
```

`oc kustomize` results in an error:

```
Mon Mar 25 04:28:09 (jslagle@teletran-1) [~/code/osp-ng/openstack-k8s-operators/dataplane-operator/examples/nova_cell_custom] (kustomizeupdate)$ oc kustomize .
error: accumulating resources: accumulation err='accumulating resources from '../../config/services/dataplane_v1beta1_openstackdataplaneservice_nova.yaml': security; file '/home/jslagle/code/osp-ng/openstack-k8s-operators/dataplane-operator/config/services/dataplane_v1beta1_openstackdataplaneservice_nova.yaml' is not in or below '/home/jslagle/code/osp-ng/openstack-k8s-operators/dataplane-operator/examples/nova_cell_custom'': must build at directory: '/home/jslagle/code/osp-ng/openstack-k8s-operators/dataplane-operator/config/services/dataplane_v1beta1_openstackdataplaneservice_nova.yaml': file is not directory
```

Is there a way we could make this work?

Reply:
Ok, so this works with the diff above:

`oc kustomize --load-restrictor LoadRestrictionsNone .`

I think we need to get rid of the file duplication and document it this way, unless there are other options?

Reply:
I removed the duplication of the nodeset and deployment samples, now all the kustomize examples needs this, I started this: https://github.com/openstack-k8s-operators/install_yamls/pull/787